src/libertem/common/shape.py

```diff
@@ -151,6 +151,42 @@ def __eq__(self, other):
         values_eq = tuple(self) == tuple(other)
         return dims_eq and values_eq
 
+    def __add__(self, other):
+        """
+        Addition is supported between a Shape object and a tuple.
+        Either the navigation or the signal dimensions of the Shape object must be zero
+        and the tuple is added to the non-zero dimension.
+        """
+        if isinstance(other, tuple):
+            if self._sig_dims != 0 and len(self._nav_shape) != 0:
+                raise ValueError("To add a Shape object to a tuple, either the navigation "
+                                 "or the signal dimensions of the Shape object must be zero")
+            elif self._sig_dims == 0 and len(self._nav_shape) == 0:
+                raise ValueError("Shape object is empty")
+            elif self._sig_dims == 0:
+                return Shape(self._nav_shape + other, sig_dims=0)
+            else:
+                return Shape(self._sig_shape + other, sig_dims=len(self._sig_shape + other))
+        else:
+            return NotImplemented
```

Comment:
Is it possible to somehow not have duplication in `__add__` and `__radd__`? Maybe by calling a helper method to do most of the actual work and validation, and just call it with different parameters in add and radd?

Reply:
Thanks, I noticed the repetition but wasn't sure how to fix it. I made a helper function called `_add_tuple_helper` - is that a good name?

Reply:
Sure! Naming things is most of the times subject to [the bikeshedding problem](https://en.wiktionary.org/wiki/bikeshedding), so for internal names, as long as a name is not misleading, I tend not to care as much, as it can also be changed later. For public APIs, where the name is fixed for a long time, it is more important to get it right, of course :grin:

Reply:
This is good to know, thanks!